addons/hr_contract/wizard/hr_departure_wizard.py

```diff
@@ -8,29 +8,45 @@
 class HrDepartureWizard(models.TransientModel):
     _inherit = 'hr.departure.wizard'
 
-    def _get_employee_departure_date(self):
-        employee = self.env['hr.employee'].browse(self.env.context['active_id'])
+    def _get_employee_departure_date(self, employee):
         if employee.contract_id.state == "open":
             return False
         expired_contract = self.env['hr.contract'].search([('employee_id', '=', employee.id), ('state', '=', 'close')], limit=1, order='date_end desc')
         if expired_contract:
             return expired_contract.date_end
-        return super()._get_employee_departure_date()
+        return super()._get_employee_departure_date(employee)
 
     set_date_end = fields.Boolean(string="Set Contract End Date", default=lambda self: self.env.user.has_group('hr_contract.group_hr_contract_manager'),
         help="Set the end date on the current contract.")
 
     def action_register_departure(self):
         """If set_date_end is checked, set the departure date as the end date to current running contract,
         and cancel all draft contracts"""
-        current_contract = self.sudo().employee_id.contract_id
-        if current_contract and current_contract.date_start > self.departure_date:
+        employees = self.employee_ids
+        Contract = self.env['hr.contract'].sudo()
+        base_query = Contract._search([('employee_id', 'in', employees.ids)])
+        current_query = Contract._search([('id', 'in', employees.contract_id.ids)])
+        if Contract.search_count([
+            ('id', 'in', current_query),
+            ('date_start', '>', self.departure_date)],
+            limit=1,
+        ):
```

Comment:
I'm not sure why they would be in the cache? 
My idea behind current_query was that it's a factoring of a domain that gets searched twice as part of 2 different bigger domains - I thought this is to optimize the second search because the results will be memoized.
I agree `base_query` ended being used only once and is not needed.